flux/src/main.rs

```diff
@@ -4,7 +4,7 @@ use std::{env, io, process::exit};
 const CMD_RUSTC: &str = "rustc";
 
 fn main() -> io::Result<()> {
-    logger::install()?;
+    let resolve_logs = logger::install()?;
```

Comment:
I feel like the proper thing to do here is to extract the `Dispatch` used and then pass it to a function which resolves the logs. However, the full type of the `Dispatch` was like two lines long and I didn't feel particularly enthused about putting that into the return type of `logger::install()`. So instead I returned a closure that would do the cleanup - this closure is to be called after the main work is done.

Reply:
I think that's fine.